internal/replacements/replacements.go

```diff
@@ -189,13 +189,8 @@ func instanceNeedsRemoval(cluster *fdbtypes.FoundationDBCluster, pod *corev1.Pod
 	}
 
 	expectedNodeSelector := cluster.GetProcessSettings(processClass).PodTemplate.Spec.NodeSelector
-	if !equality.Semantic.DeepEqual(pod.Spec.NodeSelector, expectedNodeSelector) {
-		logger.Info("Replace instance",
-			"reason", fmt.Sprintf("nodeSelector has changed from %s to %s", pod.Spec.NodeSelector, expectedNodeSelector))
-		return true, nil
-	}
 
-	if cluster.Spec.UpdatePodsByReplacement {
+	if cluster.Spec.UpdatePodsByReplacement || !equality.Semantic.DeepEqual(pod.Spec.NodeSelector, expectedNodeSelector) {
```

Comment:
Can we still leave the node selector check in a different if statement? That makes it easier to see why the Pod was replaced with this change we will only see "specHash has changed from ... to ..." but we can be more precise by logging that the node selector doesn't match.